kpops/components/base_components/base_defaults_component.py

```diff
@@ -82,7 +81,8 @@ class BaseDefaultsComponent(DescConfigModel, ABC):
     )
 
     def __init__(self, **values: Any) -> None:
-        if values.get("enrich", True):
+        pipeline_path = ENV.get("pipeline_path", None)
+        if values.get("enrich", True) and pipeline_path:
```

Comment:
imo the check should be in `extend_with_defaults`. that way, we can still run substitution